### PR TITLE
MM-18089 - Custom setting type manifest documentation

### PR DESCRIPTION
**Pull request by marianunez**
#### Summary

Added new `custom` setting type to plugin manifest documentation that is now supported by this webApp PR https://github.com/mattermost/mattermost-webapp/pull/3941

#### Ticket Link
[MM-18089](https://mattermost.atlassian.net/browse/MM-18089)

### CODE REVIEW

**Comment by marianunez** [comment]
@amyblais Given that the [custom admin setting feature](https://github.com/mattermost/mattermost-webapp/pull/3941) for plugins is going out in 5.18, this documentation change should be cherry-picked to 5.18.

File: model/manifest.go [hunk 1]
```diff
@@ -50,6 +50,8 @@ type PluginSetting struct {
 	// "longtext" will result in a multi line string that can be typed in manually.
 	//
 	// "username" will result in a text setting that will autocomplete to a username.
+	//
+	// "custom" will result in a custom defined setting and will load the custom component registered for the webApp System Console.
```
Comment (hanzei) [inline model/manifest.go]: `webApp` or `webapp`?

Comment (marianunez) [reply model/manifest.go]: Or `Web App`? It seems to be referenced that way in docs right now: https://developers.mattermost.com/extend/plugins/webapp/reference/

Will updated.

**Pull request merged**